libraries/protocol/include/scorum/protocol/ref.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+#include <memory>
+
+namespace scorum {
+namespace utils {
+
+/*
+ * std::reference_wrapper has special semantics in template argument deduction rules so it doesn't work properly with
+ * fc::static_variant.
+ *
+ * Using custom 'utils::ref' type instead which doesn't have such semantics
+ */
+template <typename T> class ref
+{
+public:
+    ref() = default;
```

Comment:
Default contractor for reference wrapper must be deleted.
Write TODO comment that it is temporary decision for current fc::variant implementation.